File: Problema06/Java/Main.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.text.DecimalFormat;

class Main {
  public static void main(String[] args) {
    // Vars / Cons + Objects
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    DecimalFormat df = new DecimalFormat("#,##0.00");

    final String PIN = "2020";
    String inputPin = null;
    double amount = 0, balance = 1000;

    try{
      // Inputs
      print("PIN: ");
      inputPin = br.readLine();

      // Process & Outputs
      if(inputPin.equals(PIN)){
        print("\nIngrese cantidad a retirar: ");
        amount = Double.parseDouble(br.readLine());

        if(amount <= balance){
          balance -= amount;
          print("Saldo: " + df.format(balance));
        }else{
          print("\nSaldo insuficiente.\nNo se realizó ninguna transacción.");
        }
      }else{
        print("\n¡PIN incorrecto!");
      }
    }catch(Exception e){
      print("Se generó la siguiente excepción: " + e.getMessage());
    }
  }

  public static void print(String s){
    System.out.print(s);
  }
}

File: Problema06/Python/main.py
# Import modules
import const

# Variables
amount = 0
balance = 1000
m_format = "{:,.2f}".format

# Inputs
try:
  pinInput = input("PIN: ")
except Exception as e:
  print("\nSe produjo la siguiente excepción: " + str(e))

# Process & Outputs
if pinInput == const.PIN:
	amount = float(input("Monto a retirar: "))
	
	if amount <= balance:
		balance -= amount
		print(f"Saldo: {m_format(balance)}")
	else:
		print("Saldo insuficiente.\nNo se realizó ninguna transacción.")
else:
	print("¡PIN incorrecto!")